toolchains/local_java_repository.bzl

```diff
@@ -191,19 +191,39 @@ def _local_java_repository_impl(repository_ctx):
 
     if not java_bin.exists:
         # Java binary does not exist
+        message = ("Cannot find Java binary {java_binary} in {java_home}; either correct your JAVA_HOME, " +
+                   "PATH or specify Java from remote repository (e.g. " +
+                   "--java_runtime_version=remotejdk_11)")
         repository_ctx.file(
             "BUILD.bazel",
-            _NOJDK_BUILD_TPL.format(
+            _AUTO_CONFIG_ERROR_BUILD_TPL.format(
                 local_jdk = local_java_runtime_name,
-                java_binary = _with_os_extension(repository_ctx, "bin/java"),
-                java_home = java_home,
+                message = repr(message.format(
+                    java_binary = _with_os_extension(repository_ctx, "bin/java"),
+                    java_home = java_home,
+                )),
             ),
             False,
         )
         return
 
     # Detect version
     version = repository_ctx.attr.version if repository_ctx.attr.version != "" else _detect_java_version(repository_ctx, java_bin)
+    if version == None:
+        # Java version could not be detected
+        message = "Cannot detect Java version of {java_binary} in {java_home}; make sure it points to a valid Java executable"
+        repository_ctx.file(
```

Comment:
It might be worth extractor a helper function to share this with the copy above, I think the only different here is the `message`?